packages/core/src/envelope.ts

```diff
@@ -109,6 +114,15 @@ export function createSpanEnvelope(spans: SentrySpan[]): SpanEnvelope {
     sent_at: new Date().toISOString(),
     ...(dscHasRequiredProps(dsc) && { trace: dsc }),
   };
-  const items = spans.map(span => createSpanEnvelopeItem(spanToJSON(span)));
+
+  const beforeSend = client && client.getOptions().beforeSendSpan;
+  let items: SpanItem[];
+
+  if (beforeSend) {
+    items = spans.map(span => createSpanEnvelopeItem(beforeSend(spanToJSON(span) as SpanJSON)));
+  } else {
+    items = spans.map(span => createSpanEnvelopeItem(spanToJSON(span)));
+  }
```

Comment:
I'm not even sure if there is a difference, but feels like it _could_ be a micro-bundlesize-improvement: What if we do this in a single loop, something like this:

```js
const items = spans.map(span => {
  const spanJson = spanToJSON(span) as SpanJSON;
  return beforeSend ? beforeSend(spanJson) : spanJson;
});
```

added benefit is that we save a `let items` that we need to re-assign 🤔 but no strong feelings, it probably doesn't matter either way.

Reply:
I was considering that but thought it wastes cycles checking the same thing over and over again. Not sure how big `span` gets tho. Thoughts?

Reply:
yeah, you're probably right! 

What about a combination of both:

```ts
const convertToSpanJson = beforeSendSpan 
  ? (span: Span) => beforeSendSpan(spanToJSON(span) as SpanJSON)) 
  : (span: Span) => spanToJSON(span) as SpanJSON;
const items = spans.map(convertToSpanJson);
```

Reply:
Yea nice, fixed.